src/opsinputs/opsinputs_varobswriter_mod.F90

```diff
@@ -963,6 +963,9 @@ subroutine opsinputs_varobswriter_populateobservations( &
       ! TODO(someone): handle this varfield
       ! call Ops_Alloc(Ob % Header % AMSUb_Temp, "AMSUb_Temp", Ob % Header % NumObsLocal, Ob % AMSUb_Temp)
     case (VarField_cloud)
+      call opsinputs_fill_fillelementtype2dfromnormalvariablewithlevels( &
+        Ob % Header % Cloud, "Cloud", Ob % Header % NumObsLocal, Ob % Cloud, &
+        ObsSpace, self % modlevs, "Cloud_", "DerivedObsValue", self % GeoVaLsAreTopToBottom, "CloudError_", "DerivedObsError")
```

Comment:
Do you have to use `DerivedObsError` here or can you just use `ObsError`? Can you remind me how the error is assigned please?

Also, I would prefer the name of the error variable to just be `Cloud` (or whatever you choose) so it matches up with the equivalent `ObsValue`.

Reply:
I think it does need to be `DerivedObsError` for the same reason that the ObsValue `Cloud` needs to be `DerivedObsValue` - these values are what are calculated in the ObsFunction covered in https://github.com/JCSDA-internal/ufo/pull/2824 which is called from the yaml. The variable is created there, but possibly set up in the yaml before calling that ObsFunction. I had assumed that ObsError acted in the same way as ObsValue in that it cannot be altered in the yaml.

I think it makes sense to do as you suggest for the name, I'll make that change here (and in our SurfaceCloud yaml @fabien-mo - I think these are the only places that would need to be changed?)

Reply:
Unlike ObsValue, you can assign to errors in a yaml (using e.g. DrawObsErrorFromFile). However, given you used DerivedObsError in the ufo PR, I don't think there's a problem keeping it as-is. In fact, it might be better to used DerivedObsError because that makes it clear that the error is not stored in the internal `obserror_` data structure.

Ultimately, as with the name of the Cloud variable, as long as everything works from ODB ingestion to Varobs/CX production then I think that's fine.